src/squlearn/encoding_circuit/qcnn_encoding_circuit.py

```diff
@@ -0,0 +1,617 @@
+import numpy as np
+from typing import Union
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector
+from qiskit.converters import circuit_to_gate, circuit_to_instruction
+
+from squlearn.encoding_circuit.encoding_circuit_base import EncodingCircuitBase
+from squlearn.encoding_circuit.circuit_library.param_z_feature_map import ParamZFeatureMap
+from squlearn.observables import CustomObservable
+
+
+class QCNNEncodingCircuit(EncodingCircuitBase):
+
+    """
+    Instantiate the object for building a QCNN.
+    In order to plot your circuit (called e.g. mycircuit) at any given point use: mycircuit.draw().
+    One can hand over the optional arguments "mpl" to get the circuit in matplotlib drawing and "decompose = True"
+    to resolve the individual components of the circuit building blocks
+    (e.g. mycircuit.draw("mpl", decompose = True)).
```

Comment:
This description is a bit unnecessary. I think this section should rather be used to describe the Idea of QCNNs. Also make references to the paper(s) that introduced the concept.